fix: Get dependency version [skip tests]

hpohekar opened the pull request:
closes https://github.com/ansys/pyfluent/issues/3835

Automate ansys-tools-report.

seanpearsonuk commented on src/ansys/fluent/core/report.py:
```diff
@@ -55,20 +57,20 @@
+    "pandas": version("pandas"),
+    "pyansys-tools-report": version("pyansys-tools-report"),
+    "pyyaml": version("pyyaml"),
 }
```
@hpohekar 
```Python
package_names = [
    "ansys-api-fluent",
    "ansys-platform-instancemanagement",
    "ansys-tools-filetransfer",
    "ansys-units",
    "docker",
    "grpcio",
    "grpcio-health-checking",
    "grpcio-status",
    "h5py",
    "nltk",
    "numpy",
    "pandas",
    "pyansys-tools-report",
    "pyyaml",
]

dependencies = {pkg: version(pkg) for pkg in package_names}
```

hpohekar replied:
Done. Thank you.

seanpearsonuk replied:
@hpohekar The following is a potentially fully automated approach, ensuring we never need to manually update this code. Note that tomllib is now part of the standard library (Python 3.11+), whereas the prior tomli would have added an extra dependency.

**AI has filled in the details here, and while the handling of individual TOML lines might still seem a bit hacky, I have kept this separate from the simpler refactoring above. However, automating the report could be a great improvement to ensure accurate and up-to-date output.**

```Python
import pathlib

# Ensure compatibility with different Python versions
try:
    import tomllib  # Python 3.11+
except ModuleNotFoundError:
    import tomli as tomllib  # Python 3.10 or earlier

def get_project_dependencies():
    """Reads pyproject.toml and extracts dependencies dynamically."""
    # Locate project root (assuming script is in 'src/ansys/fluent/core/')
    project_root = pathlib.Path(__file__).resolve().parents[3]
    pyproject_path = project_root / "pyproject.toml"

    # Read the TOML file
    with open(pyproject_path, "rb") as f:
        pyproject_data = tomllib.load(f)

    # Extract dependencies from the 'project' section
    return pyproject_data.get("project", {}).get("dependencies", [])

def get_dependency_versions():
    """Builds a dictionary of dependencies mapped to their installed versions."""
    package_versions = get_project_dependencies()

    return {
        pkg.split(">")[0].split("<")[0].split("~")[0]: version(pkg.split(">")[0].split("<")[0].split("~")[0])
        for pkg in package_versions
    }

# Generate dependencies dictionary dynamically
dependencies = get_dependency_versions()

# Example usage:
if __name__ == "__main__":
    print("Extracted dependencies:", dependencies)
```

hpohekar replied:
@seanpearsonuk 

Great! Thanks a lot. 

Actually copilot filled `version(<package-name>)` everywhere quickly therefore I hadn't thought of automation.

hpohekar commented:
> @hpohekar Will the following work?
> 
> ```
> from importlib.metadata import requires
> from packaging.requirements import Requirement
> dependency_versions = {}
> for requirement in requires("ansys-fluent-core"):
>     split_extra = requirement.split(" ; ")
>     if len(split_extra) == 1 or (len(split_extra) == 2 and split_extra[1].split(" == ")[1] == '"reader"'):
>         req = Requirement(split_extra[0])
>         dependency_versions[req.name] = str(req.specifier)
> 
> from pprint import pprint
> pprint(dependency_versions)
> ```
> 
> ```
> {'ansys-api-fluent': '>=0.3.35',
>  'ansys-platform-instancemanagement': '~=1.1',
>  'ansys-tools-filetransfer': '<0.3,>=0.1',
>  'ansys-units': '<0.5,>=0.3.3',
>  'docker': '>=7.1.0',
>  'grpcio': '>=1.30.0',
>  'grpcio-health-checking': '>=1.30.0',
>  'grpcio-status': '>=1.26.0',
>  'h5py': '==3.12.1',
>  'lxml': '>=4.9.2',
>  'nltk': '>=3.9.1',
>  'numpy': '<3.0.0,>=1.14.0',
>  'pandas': '<2.3,>=1.1.0',
>  'pyansys-tools-report': '>=0.8.1',
>  'pyyaml': '>=6.0'}
> ```
> 
> The module `packaging` is used in our test/doc environment. If it is not available when report.py is executed, we can split `split_extra[0]` by the comparison operator of [version specifiers](https://packaging.python.org/en/latest/specifications/version-specifiers/#id5).

@mkundu1 Done. Thank you.
